### ISSUE TITLE
Prediction is slower when model is loaded than if it is fited during the process

### ISSUE THREAD

**Issue by AxelRagobert** [tflearn/tflearn#1098]
Hello,

I have a strange issue, the DNN.predict method is quite slower when I load my model's weight than when I train with the fit method. I've also noted that when I run a prediction over a batch of images, it's getting faster and faster to predict.

Here is my code 
`class Reseau(object):

def __init__(self, img_size, lr=-1, activation=" "):
        tf.logging.set_verbosity(tf.logging.ERROR)
        self.lr = lr
        self.activation = activation
        self.img_size = img_size
        self.alreadySaved = 0

def setting(self, X, Y, test_x, test_y, nbEpoch):
        tflearn.init_graph(num_cores=32, gpu_memory_fraction=1)
        with tf.device("/device:GPU:0"):
            convnet = input_data(shape=[None, self.img_size, self.img_size, 3], name='input')
            
            convnet = conv_2d(convnet, 32, 5, activation=self.activation)
            convnet = max_pool_2d(convnet, 5)

            convnet = conv_2d(convnet, 64, 5, activation=self.activation)
            convnet = max_pool_2d(convnet, 5)

            convnet = conv_2d(convnet, 128, 5, activation=self.activation)
            convnet = max_pool_2d(convnet, 5)

            convnet = conv_2d(convnet, 64, 5, activation=self.activation)
            convnet = max_pool_2d(convnet, 5)

            convnet = conv_2d(convnet, 32, 5, activation=self.activation)
            convnet = max_pool_2d(convnet, 5)
            
            convnet = flatten(convnet)
            
            convnet = fully_connected(convnet, 1024, activation=self.activation, name='last')
            convnet = fully_connected(convnet, 1024, activation=self.activation, name='last')
            convnet = fully_connected(convnet, 1024, activation=self.activation, name='last')
            convnet = fully_connected(convnet, 1024, activation=self.activation, name='last')
            convnet = dropout(convnet, 0.8)

            convnet = fully_connected(convnet, 2, activation='softmax')
            convnet = regression(convnet, optimizer='adam', learning_rate=self.lr, loss='categorical_crossentropy', name='targets')

            self.model = tflearn.DNN(convnet, tensorboard_dir='log')

            if self.alreadySaved == 0:
                self.model.fit({'input': X}, {'targets': Y}, n_epoch=nbEpoch, validation_set=({'input': test_x}, {'targets': test_y}), snapshot_step=500, show_metric=True, run_id="model")
                self.model.save("./model")
            else:
                self.model.load("./model", weights_only=True)
        return self.model

def predire(self, img, label):
        image = array(img).reshape(1, self.img_size,self.img_size,3)
        model_out = self.model.predict(image)
        rep = 0

        if np.argmax(model_out) == np.argmax(label): rep = 1
        else: rep = 0

        return rep
`

Here is a part of my main

`reseau.setting(X, Y, test_x, test_y, NB_EPOCH)

    X = np.array([i[0] for i in test]).reshape(-1,IMG_SIZE,IMG_SIZE,3)
    Y = [i[1] for i in test]
    cpt = 0

    vrai = 0
    start_time = time.time()

    for i in range(20):
        cpt = 0

        vrai = 0
        start_time = time.time()
        for img in tqdm(X):
            prediction = reseau.predire(img, Y[cpt])
            cpt += 1
            if prediction == 1:
                vrai += 1`

As you can see, I predict the same batch of images 20 times. The first time is always slower than the other ones (without fitting, I predict 82 images the first and then 340 a second, with fitting, it's 255 images the first time and 340 a seconde then).

I'm really out of idea to fix this.
